Shifty.App/Repositories/VoucherRepository.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using LanguageExt;
+using LanguageExt.Common;
+using Shifty.Api.Generated.AnalogCoreV2;
+using static LanguageExt.Prelude;
+
+namespace Shifty.App.Repositories
+{
+    public class VoucherRepository : IVoucherRepository
+    {
+        private readonly AnalogCoreV2 _client;
+
+        public VoucherRepository(AnalogCoreV2 client)
+        {
+            _client = client;
+        }
+        
+        public async Task<Either<Error, ICollection<IssueVoucherResponse>>> IssueAsync(int amount, int productId, string description)
+        {
+            var account = await _client.ApiV2AccountGetAsync();
+            var request = new IssueVoucherRequest()
+            {
+                Amount = amount,
+                Description = description,
+                ProductId = productId,
+                Requester = account.Name,
+                VoucherPrefix = account.Email[..3],
```

Comment:
Don't we want this to be set by the user, so it can relate to the product/ occasion instead? This will be part of the final voucher code after all, so I don't think we want the email

Reply:
With the current temp solution, my feeling was that I had to fill out "too many" fields (more than seems necessary). That's why I wanted it to automatically fill out this field (which I would argue is not important whether it says "AAA" or "AGU"). 

I have now changed it to have an input field that on load initialises to the first three letters of the user mail instead. This seems to be the best compromise - allowing for the configuration for the user, but removes the hassle of having to write too much myself. What do you think?

Reply:
Hmm, it might be an alright compromise, though I am unsure what the intended "business case" of having the prefix is. Perhaps @marfavi or @jonasanker can weigh in for a second, what they intended with this field. Personally, I imagined it was something to identify the batch of vouchers by, or the product it was mapping to perhaps.

Reply:
I think the prefix was to identify a group of tickets or have some familiarity with a batch of vouchers. There's no fixed rule behind it. Previous examples as far as I know have been ITD for the IT department which ordered vouchers, TUT for tutors in intro week or ESP for espresso vouchers.

Reply:
Yeah, they're a way to identify a voucher by just the voucher code itself. For some cases, the  description/issuer fields should be sufficient. One case I can imagine the prefix being relevant is for certain customers purchasing different kinds of vouchers.

Consider some customer wanting to give out different kinds of vouchers for prizes at some event. We could generate ten vouchers with prefix FCY good for a single fancy drink each, and five vouchers with prefix 10F, each good for one clip card (containing 10 fancy drink clips). With prefixes, the customer has some way of discerning between the two kinds of vouchers they are about to give out.

That's just a scenario I made up. I personally don't see much other benefit from the prefix, given that we already save more descriptive info embedded in the voucher itself.

Maybe one area of experimentation would be to always set the prefix to something relevant to the product it is good for, like the examples I gave above. This prefix could be determined by a new "voucher prefix" field in the product entity, or by simply taking the first three letters of the product name (would be less clear).

Reply:
That being said, I'd also be fine with the field being deprecated, setting it to some arbitrary value for now.

Reply:
I have put back the "Requester" field - I had misinterpreted the use case.

It might be a use case that we want to define our own prefixes, so I now changed it so it prefills the account's mail's first three letters. I think this improves the UX by having to fill less fields if I do not mind it.